tests/variable_init_test.py

```diff
@@ -354,7 +354,7 @@ def check(v):
         assert v['x', 0].value == values[0]
```

Comment:
Can you please update the requirements and remove the filtered warnings as well ??

- In `pyproject.toml`
```
  # Explained in variable_init_test.py. TODO Address this!
  'ignore:Creating an ndarray from ragged nested sequences:DeprecationWarning',
  'ignore:Creating an ndarray from ragged nested sequences:numpy.VisibleDeprecationWarning',
```

Reply:
`pip-compile-multi` cannot resolve the environment consistently at the moment. It picks numpy=1.24.1 in `base.txt` but 1.23.5 in `extra.txt`. This is because numba requires <1.24

Reply:
So we need to add the same limit for now. Unless you know of a workaround

Reply:
So what happens when that env is created? `test.txt` includes both.

Reply:
It picks whatever version is selected in the file you use. The point is that if you create an env from `base.in` (i.e. have pip resolve dependencies) and then `pip install -r extra.in`, it needs to downgrade numpy.

Reply:
Should we consider making a separate env for the numba-dependent tests?

Reply:
Let's wait a bit, maybe numba update soon enough?

Reply:
So you suggest developers downgrade to numpy=1.23? Or just go ahead with this PR while keeping CI running with 1.23?

Reply:
Merge this PR, don't update any requirements files, and revisit in 1 or 2 weeks.

Dev envs will automatically use 1.23 if you install numba and 1.24 if you don't. Scipp should continue working in either case with this PR as there is only a low chance that something else broke.

Reply:
You will probably have to reverse the removed ignored warnings?